compiler/rustc_span/src/hygiene.rs

```diff
@@ -1012,16 +1020,11 @@ pub struct HygieneEncodeContext {
 }
 
 impl HygieneEncodeContext {
-    pub fn encode<
-        T,
-        R,
-        F: FnMut(&mut T, u32, SyntaxContextData) -> Result<(), R>,
-        G: FnMut(&mut T, u32, ExpnData, ExpnHash) -> Result<(), R>,
-    >(
+    pub fn encode<T, R>(
         &self,
         encoder: &mut T,
-        mut encode_ctxt: F,
-        mut encode_expn: G,
+        mut encode_ctxt: impl FnMut(&mut T, u32, SyntaxContextData) -> Result<(), R>,
+        mut encode_expn: impl FnMut(&mut T, u32, ExpnData, ExpnHash) -> Result<(), R>,
```

Comment:
`ExpnData` is heavy, any specific reason to pass it by value?

Reply:
This is required by the encoding code in the on-disk cache, which accepts `&(ExpnData, ExpnHash)`, but not `&(&ExpnData, ExpnHash)`. However, #86676 can pass a reference.